python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -55,6 +55,22 @@ def read_partition(
         if index is not None and index[0] in df.columns:
             df = df.set_index(index[0])
 
+        if len(piece.partition_keys) > 0:
+            # Partitioning logic copied from
+            # (apache/arrow/python/pyarrow/parquet.py)
+            if partitions is None:
+                raise ValueError("Must pass partition sets")
+            import numpy as np
+            import pyarrow as pa
+
+            for i, (name, index2) in enumerate(piece.partition_keys):
+                indices = np.array([index2], dtype="i4").repeat(len(df))
+                dictionary = partitions.levels[i].dictionary
+                arr = pa.DictionaryArray.from_arrays(
+                    indices, dictionary
+                ).to_pandas()
```

Comment:
Similarly here, should we construct a categorical column directly on the GPU instead?